src/utils.jl

```diff
@@ -32,3 +32,7 @@ function catsamples_and_acc(
     return (x, ∑y)
 end
 
+function samples_expand_dim(x::AbstractVector)
```

Comment:
I was thinking of using this as an abstraction for the future, where we get to use `NamedTuple`-variate random variables. It would also make it easier to track where we explicitly assume which axes are what, which I think could be useful for maintenance.